src/coap_debug.c

```diff
@@ -646,17 +660,29 @@ coap_show_pdu(coap_log_t level, const coap_pdu_t *pdu) {
       outbuflen = strlen(outbuf);
       snprintf(&outbuf[outbuflen], sizeof(outbuf)-outbuflen,  ">>");
     } else {
-      if (print_readable(data, data_len, buf, sizeof(buf), 0)) {
-        size_t max_length;
-        outbuflen = strlen(outbuf);
-        max_length = sizeof(outbuf)-outbuflen;
-        if (snprintf(&outbuf[outbuflen], max_length,  "'%s'", buf) >= (int)max_length)
-          outbuf[sizeof(outbuf)-1] = '\000';
+      size_t max_length;
+      outbuflen = strlen(outbuf);
+      max_length = sizeof(outbuf)-outbuflen;
+      if (max_length > 1) {
+        outbuf[outbuflen++] = '\'';
+        outbuf[outbuflen] = '\000';
+        max_length--;
+      }
+      if (max_length > 1) {
+        outbuflen += print_readable(data, data_len,
+                                    (unsigned char*)&outbuf[outbuflen],
+                                    max_length, 0);
+      }
+      /* print_readable may be handling unprintables - hence headroom of 4 */
+      if (outbuflen < ((int)sizeof(outbuf))-4-1) {
```

Comment:
The typecast here is useless because the result is cast to unsigned long anyway because of `outbuflen`'s type.

Reply:
Not sure where that mental aberration came from - other than if `outbuf` size happened to be less than 5 bytes (as set by COAP_DEBUG_BUF_SIZE) unexpected memory would get stamped on.  Typecast removed and code pushed.

Reply:
Thanks. I will add a pre-processor check after merge just to be sure that this constraint is not lost over time.